test/testenv/build.go

```diff
@@ -170,6 +223,19 @@ func displaySolveStatus(ctx context.Context, t *testing.T) chan *client.SolveSta
 		}
 	})
 
+	return f
+}
+
+func displaySolveStatus(ctx context.Context, t *testing.T) chan *client.SolveStatus {
```

Comment:
IIUC, the caller would only need to read from this channel. So maybe we can enforce some type safety here?

```suggestion
func displaySolveStatus(ctx context.Context, t *testing.T) <-chan *client.SolveStatus {
	ch := make(chan *client.SolveStatus)
	done := make(chan struct{})

	output := getBuildOutputStream(ctx, t, done)
	display, err := progressui.NewDisplay(output, progressui.AutoMode, progressui.WithPhase(t.Name()))
	if err != nil {
		t.Fatal(err)
	}

```

Reply:
The caller needs `chan *client.SolveStatus`.
It gets passed to the buildkit client which is what writes to the channel.